scipy/integrate/tests/test_quadrature.py

```diff
@@ -563,20 +566,28 @@ def test_nd(self, axis, x_ndim, x_len, i_ndim, dx):
         i_shape = shape_len_1 if i_ndim == 3 else ()
 
         # initialize arguments
-        y = rng.random(size=shape)
+        y = xp.asarray(rng.random(size=shape))
         x, dx = None, None
         if dx:
             dx = rng.random(size=shape_len_1) if x_ndim > 1 else rng.random()
+            dx = xp.asarray(dx)
         else:
             x = (np.sort(rng.random(size=shape), axis=axis) if x_ndim > 1
                  else np.sort(rng.random(size=shape[axis])))
-        initial = None if i_ndim is None else rng.random(size=i_shape)
+            x = xp.asarray(x)
+        initial = None if i_ndim is None else xp.asarray(rng.random(size=i_shape))
 
         # compare results
         res = cumulative_simpson(y, x=x, dx=dx, initial=initial, axis=axis)
-        ref = cumulative_simpson_nd_reference(y, x=x, dx=dx, initial=initial, axis=axis)
-        np.testing.assert_allclose(res, ref, rtol=1e-15)
+        ref = cumulative_simpson_nd_ref(
+            np.asarray(y), x=np.asarray(x), dx=None if dx is None else np.asarray(dx),
```

Comment:
Should there be a comment explaining why we are using `np.` and `xp.` above? This seems like just partial support being added/partially converted test? `cpu_only` on its own doesn't seem sufficient